include/mull/Filters/FilePathFilter.h

```diff
@@ -18,10 +20,16 @@ class FilePathFilter : public MutationFilter,
   bool shouldSkip(llvm::Instruction *instruction) const override;
   std::string name() override;
   void exclude(const std::string &filter);
+  void include(const std::string &filter);
 
 private:
   bool shouldSkip(const mull::SourceLocation &location) const;
+  bool shouldSkip(const std::string &sourceFilePath) const;
+
+  std::vector<std::regex> includeFilters;
+  std::vector<std::regex> excludeFilters;
 
-  std::vector<std::regex> filters;
+  mutable std::unordered_map<std::string, bool> cache;
```

Comment:
I am not sure about this `mutable` but otherwise we have to remove `const` from all of the declarations. I am fine with both options because this `FilePathFilter` class is very small is clear.